Candidate deploy view

dsingleton opened the pull request:
Simplify deployment process by giving you one page with all
the links and information you should need to do a deploy.

- Navigate to the new view from the list of commits/tags for
  an application
- See where production is, what release and how old
- Preview what commits, and authors, this release will contain
  with a link to view the full file diff on github
- Deep link to the staging/production deploy jobs, prefilling
  the application and release tag
- Bonus tweaks to some app commit listing UI

Very much minimum viable improvement, it could and will get better.

![image](https://cloud.githubusercontent.com/assets/63201/5860379/eafe8564-a258-11e4-858a-0354b947fe13.png)

Picking a candidate release:
![image](https://cloud.githubusercontent.com/assets/63201/5860444/3e8a4aec-a259-11e4-8455-b415c8cbc087.png)
(This isn't very intuitive, but i've got plans to improve this view separately)

fatbusinessman commented on app/views/applications/deploy.html.erb:
```diff
@@ -0,0 +1,93 @@
+
+<main>
+  <h2>Candidate Release <span class="label label-info"><%= @release_tag %></span></h2>
+  <p class="lead add-top-margin">Production was deployed to <span class="label label-danger"><%= @production_deploy.version %></span><%= time_ago_in_words(@production_deploy.created_at) %> ago</p>
```
The phrasing on this feels off: I always think of this as a version being deployed to production, not production being deployed to a version. I would suggest rephrasing this to “`release_x` was deployed to production _n_ days ago”, but I’m guessing the alignment of the two tags is intentional. Not sure what other options there are.

dsingleton replied:
You're right, that reads badly. Leading with the tag doesn't read quite right either, how about: 

> Production is release_2156 — deployed at 11:59am on Monday

(I've also changed the date/time format to have more granularity and match the application list.)

fatbusinessman commented:
From a quick look, I very much like this. I’m next in the office on Tuesday: unless someone else has reviewed and merged it by then, I’ll give it a more thorough review.

fatbusinessman commented:
It doesn’t look like the author images are working at the moment:

![Broken author images](https://cloud.githubusercontent.com/assets/32775/5938917/44d02988-a6f7-11e4-8f2a-e9ef7c8055ff.png)

This is on preview and on my dev machine: the `<img>` tags appear to be coming through with an empty `src`.

fatbusinessman commented:
When someone writes full, multi-paragraph commit messages (as with more involved applications like Rummager), the list of commits gets very long and unwieldy. Given there’s a link through to the commit page on GitHub, I wonder whether that would be better left out entirely.

dsingleton commented:
Comments make sense. I'd be tempted to bring it back in the future as toggle-able, but i'm removing it for now.

Also fixed images, and updated preview.

fatbusinessman commented:
> I'd be tempted to bring it back in the future as toggle-able […]

Yup: good idea :+1:

fatbusinessman commented:
This is really nice work, @dsingleton, especially the way it integrates with Jenkins. As one of the people who will be deploying things with it. thank you.

benilovj commented:
@dsingleton really awesome work, just used this for the first time, my deploy was done in about a minute!
